python/graphstorm/dataloading/sampler.py

```diff
@@ -389,3 +392,247 @@ def sample_blocks(self, g, seed_nodes, exclude_eids=None):
+    dataset_path : str
+        Path to the data files.
+        Mutually exclusive with :attr:`files`.
+    """
+
+    def __init__(self, dataset_path=None):
+        self.dataset_path = None
+        self.files = None
+
+        if not dataset_path:
+            raise ValueError(
+                "Dataset_path={}, should be specified.".format(
+                    dataset_path
+                )
+            )
+        else:
+            if not isinstance(dataset_path, str):
+                raise TypeError(
+                    "dataset_path should be a str, but got " "dataset_path={}".format(dataset_path)
+                )
+
+            self.dataset_path = dataset_path
+            self.files = [
+                os.path.join(dataset_path, f)
+                for f in os.listdir(dataset_path)
+                if os.path.isfile(os.path.join(dataset_path, f))
+            ]
+            self.files = [f for f in self.files if os.path.getsize(f)>0]
+            self.files.sort()
+
+            if len(self.files)==0:
+                raise ValueError (
+                    "no non-empty files found at top directory {}.".format(dataset_path))
+
+        self.num_files = len(self.files)
+        print ("found {} files from {}".format(self.num_files, dataset_path))
+
+    def __len__(self):
+        return self.num_files
+
+    def __iter__(self):
+        raise NotImplementedError
+
+    def __next__(self):
+        raise NotImplementedError
+
+    def to_json_obj(self):
+        return {
+            "files": self.files,
+            "sampler_class": self.__class__.__name__,
+        }
+
+class SequentialFileSampler():
+    r"""Sequential File Sampler. Samples file sequentially.
+
+    Parameters:
+    ----------
+    file_indices : list of int
+        File indices for a local trainer
+    is_train : bool
+        Set to ``True`` if it's training set.
+    """
+
+    def __init__(self, file_indices=None, is_train=True):
+        self.file_indices = file_indices
+        self.num_local_files = len(self.file_indices)
+        self._indices = list(range(self.num_local_files))
+        self._index = -1
+        self.reset = True
+        self.is_train = is_train
+
+    def __iter__(self):
+        if self.reset:
+            self._index = -1
+        self.reset = True
+        return self
+
+    def __next__(self):
+        """ Get index for next file"""
+        self._index += 1
+
+        if self._index >= self.num_local_files:
+            if self.is_train:
+                self._index = 0
+            else:
+                # non-infinite sampler if it's for evaluation.
+                self._index = 0
+                return None
+
+        return self.file_indices[self._index]
+
+class RandomShuffleFileSampler():
+    r"""Random File Sampler. Has files reshuffled for sampling.
+
+    Parameters:
+    ----------
+    file_indices : list of int
+        File indices for a local trainer
+    """
+
+    def __init__(self, file_indices=None):
```

Comment:
Can file_indices be None?
Why there is no is_train argument?

Reply:
RandomShuffleFileSampler can only be used for training.